### DISCUSSION TITLE
📔 Project documentation for end-user

### DISCUSSION THREAD

**Discussion by xeho91** [svelte-docgen/svelte-docgen#33] [Ideas]
Given several packages we are about to have. We'll need something to wrap together for end-user guides and documentation on how to get started using it and what his options are.

We already are using [TypeDoc](https://github.com/TypeStrong/typedoc), but it's main purpose in my mind was to help contributors & maintainers navigate through this project repository.

---

**Question: What are the available options in the ecosystem to create this kind of a documentation site?**

## Requirements

- [Svelte](https://github.com/sveltejs/svelte) used, and of course with `v5` support, hoping to boost/give the reason for the potential collaboration(s) our Svelte ecosystem,
- actively maintained,
- markdown supported.

## Candidates

- [SveltePress](https://github.com/svelte-docgen/svelte-docgen/discussions/33#discussioncomment-11576524)

**Comment by xeho91** [comment]
I'm currently looking at: https://github.com/SveltePress/sveltepress

It seems to fit all the criterias.

**Reply by ciscorn** [reply]
SveltePress seems like the best option to me as well.

**Reply by ciscorn** [reply]
I also find it appealing that SveltePress is just an extension build on top of SvelteKit.

**Reply by ciscorn** [reply]
However, I think there might be some challenges in extending themes, especially when trying to add something like a “Playground.”

**Reply by xeho91** [reply]
How blocking it is?

I don't have the access to the phone, but I remember that sidebar on the left does eat some space for the playground. Plus the paddings/margins in the content body. So I share the concern of that's the case.

Also, couldn't we for example do some styling overrides using `*.svelte` extension _(which we will have to use anyway)_ instead of `*.md`?

**Reply by ciscorn** [reply]
Yes, I share exactly the same concerns.

Since `+page.svelte` files are also, in principle, displayed within the main content area, I imagine we’d need some hacky styling overrides to make them fully occupy the `<main>` area.

Deciding whether to build the docs with bare SvelteKit, on top of SveltePress, or through other approaches is a tough call. That said, we can always change the direction later if needed.

**Comment by ciscorn** [comment]
Where should we host the user documentation?

- **GitHub Pages**
  - Monorepo setup in this repository (Currently used for TypeDoc-generated documentation at the root.)
  - Organization repository (Host on the organization’s page: https://svelte-docgen.github.io/)
- **Third-party platforms**
  - Vercel, Cloudflare Pages, etc. (Easy to set up, supports PR previews, but free plans have limitations.)
    - https://vercel.com/guides/can-vercel-sponsor-my-open-source-project

**Reply by xeho91** [reply]
### Separate repository or inside monorepo?

I think for starter, we could put inside the monorepo under the `./apps/documentation` _(name is an example, can be shortened)_.

I can't see how this decision could burden us in the future. I can be persuaded though, because I don't have a strong preference, not experience where this infrastructure could be a PITA.
In example of Svelte, I believe the reason they have separated repository for their documentation & playground and tutorial _(svelte.dev)_ is that packages: `svelte`, `@sveltejs/kit` and `@sveltejs/cli` are in separate repositories.


### Hosting options

No strong preference, other than truly supporting OSS initiatives and hopefully easy to setup preview for PRs on GitHub.

I haven't used Cloudfare Pages in the past, so I can't speak for their ease-of-integration. I used Vercel, is ok.
If you prefer the first one _(CP)_, I wouldn't mind! Because I will have a chance to learn how it works internally.

**Reply by ciscorn** [reply]
I agree with keeping the documentation within the monorepo. It makes collaboration and discussions much easier to manage.

As for the hosting platform, I don’t have a strong preference either. Both Vercel and Cloudflare are well-aligned with the OSS community.

Vercel previously offered a [sponsorship program](https://vercel.com/guides/can-vercel-sponsor-my-open-source-project) for OSS projects, though it’s currently being revamped.

Cloudflare Pages/Workers, while significantly cheaper on paid plans (just $5/month), has stricter limitations on the free plan (e.g., up to 500 builds per month). They also appear to have an OSS support initiative called [Project Alexandria](https://www.cloudflare.com/en-us/lp/project-alexandria/), though I’m not too familiar with the details.

In any case, whichever option we go with, we can always migrate later if needed.

**Reply by ciscorn** [reply]
@xeho91 For now, I’m thinking of going with Vercel.

Without their OSS initiative support, the “up to 500 builds per month” limit on CF Pages could be a challenge.

**Reply by xeho91** [reply]
Aye, Vercel is it then! 🟢
